File: Discrete_Math/lab2/task2.cpp
#include <iostream>

float a(int n, float vals[]) {
	if (n == 0)
		return 1.0;
	if (n == 1)
		return -7.0;

	if (!vals[n])
		vals[n] = 5*a(n-1, vals) + 6*a(n-2, vals);

	return vals[n];
}

int main() {
	
	float values[100] = {0};
	values[0] = 1.0, values[1] = -7.0;

	std::cout << "a(100) = " << a(100, values) << std::endl;

	return 0;
}

File: Discrete_Math/lab2/task2.py
def a(n, vals):
    if n == 0:
        return 1
    if n == 1:
        return -7

    if not vals[n]:
        vals[n] =  5*a(n-1, vals) + 6*a(n-2, vals)

    return vals[n]


def main():
    values = [0 for i in range(0, 101)]

    values[0] = 1
    values[1] = -7

    n = int(input("Введите n для нахождения a(n):\n"))
    print(f"a({n}) =", a(n, values))


if __name__  == "__main__":
    main()
